src/plugins/visualizations/public/legacy/build_pipeline.ts

```diff
@@ -490,8 +490,8 @@ export const buildPipeline = async (
   }
 ) => {
   const { indexPattern, searchSource } = vis.data;
-  const query = searchSource!.getField('query');
```

Comment:
this is used for the kibana_context function, and should be irrelevant to the visualizaiton. tsvb function can ignore those if it wishes to, but shouldnt in my opinion, as we want it to work outside of visualize context as well.

so i think the actual problem is why is there a searchSource query set for tsvb saved objects, if it shouldn't be as there is no query bar in tsvb application ?

we should probably try to address that problem directly.

Reply:
> this is used for the kibana_context function, and should be irrelevant to the visualizaiton

In case of TSVB, the visualization is using the `kibana_context` passed in as input to do its data fetching. It needs to do so to get filter and query of the current dashboard and time range, but it shouldn't use the search source of the saved object, because it should be empty anyway.

I agree this is not a super clean fix, but it's meant as a temporary measure to fix the referenced bug till we can remove `build_pipeline` completely and make tsvb code responsible for its complete expression. When we do this, we can just not use `kibana_context` at all and choose to not deserialize the saved search source (right now this is happening automatically for all visualizations).

> so i think the actual problem is why is there a searchSource query set for tsvb saved objects, if it shouldn't be as there is no query bar in tsvb application ?

I agree it shouldn't, but for now it has to have one because that's how the visualization saved object is structured and in some exotic cases (multiple updates, auto generated saved objects) actual data ends up in there, causing the referenced bug.

Reply:
current plan was to not give tsvb control over that part of expression, visualize frame/embeddable would handle that internally.

any idea why we have those queries on the saved objects ? (why its not empty for tsvb) ?
1. lets make sure we fix that (so new tsvb panels no longer have that problem)
2. we can write a simple migration script (or migrate on the fly inside update_vis_state) to remove those if we know that none of saved tsvbs should have those set.